tests/serving-functions-go.test.js

```diff
@@ -0,0 +1,104 @@
+const test = require('ava')
+const pWaitFor = require('p-wait-for')
+
+const { withDevServer } = require('./utils/dev-server')
+const got = require('./utils/got')
+const { withSiteBuilder } = require('./utils/site-builder')
+
+const WAIT_INTERVAL = 1800
+const WAIT_TIMEOUT = 30000
+
+test('Updates a Go function when a file is modified', async (t) => {
```

Comment:
Feels a bit weird to be including the Go source files in the test file, but I'm not sure how else to do it.

Reply:
We can have a subdirectory `go-functions` and read the code from there?
The benefit is that we'll get code formatting and highlighting.

This is good too 🚀